File: src/app/modules/startup/login/selected_account.nim
import NimQml
import item

QtObject:
  type SelectedAccount* = ref object of QObject
    item: Item

  proc setup(self: SelectedAccount) =
    self.QObject.setup

  proc delete*(self: SelectedAccount) =
    self.QObject.delete

  proc newSelectedAccount*(): SelectedAccount =
    new(result, delete)
    result.setup

  proc setSelectedAccountData*(self: SelectedAccount, item: Item) =
    self.item = item

  proc getName(self: SelectedAccount): string {.slot.} = 
    return self.item.getName()

  QtProperty[string] username:
    read = getName

  proc getIdenticon(self: SelectedAccount): string {.slot.} =
    return self.item.getIdenticon()

  QtProperty[string] identicon:
    read = getIdenticon

  proc getKeyUid(self: SelectedAccount): string {.slot.} =
    return self.item.getKeyUid()

  QtProperty[string] keyUid:
    read = getKeyUid

  proc getThumbnailImage(self: SelectedAccount): string {.slot.} = 
    return self.item.getThumbnailImage()

  QtProperty[string] thumbnailImage:
    read = getThumbnailImage

  proc getLargeImage(self: SelectedAccount): string {.slot.} = 
    return self.item.getLargeImage()

  QtProperty[string] largeImage:
    read = getLargeImage

File: src-cpp-structure/projects/App/Modules/Startup/Login/SelectedAccount.cpp
#include "SelectedAccount.h"

using namespace Status::Modules::Startup::Login;

SelectedAccount::SelectedAccount(QObject* parent)
    : QObject(parent)
{
}

void SelectedAccount::setSelectedAccountData(const Item& item)
{
    m_item = item;
}

QString SelectedAccount::getName()
{
    return m_item.getName();
}

QString SelectedAccount::getIdenticon()
{
    return m_item.getIdenticon();
}

QString SelectedAccount::getKeyUid()
{
    return m_item.getKeyUid();
}

QString SelectedAccount::getThumbnailImage()
{
    return m_item.getThumbnailImage();
}

QString SelectedAccount::getLargeImage()
{
    return m_item.getLargeImage();
}
